webapp/app.py

```diff
@@ -0,0 +1,92 @@
+# Code
+import datetime
+import glob
+import json
+import os
+
+# Packages
+import flask
+from canonicalwebteam.flask_base.app import FlaskBase
+from canonicalwebteam.templatefinder import TemplateFinder
+
+
+# Rename your project below
+app = FlaskBase(
+    __name__,
+    "docs.vanillaframework.io",
+    template_folder="../docs",
+    static_folder="../docs/static",
+    template_404="404.html",
+    template_500="500.html",
+)
+
+
+# Helpers
+# ===
+def _get_examples():
+    example_files = glob.glob("docs/examples/*/**/*.html", recursive=True)
+    examples = {}
+
+    for filepath in sorted(example_files):
+        # Remove "docs/examples/" prefix
+        prefix_length = len("docs/examples/")
+        example_path = filepath[prefix_length:]
+        # Remove extension
+        example_path = os.path.splitext(example_path)[0]
+
+        name_parts = example_path.split(os.sep)
+        outermost_parent = name_parts.pop(0)
+
+        # Make path parts sentence case
+        friendly_parts = []
+        for part in name_parts:
+            friendly_parts.append(
+                part.capitalize().replace("-", " ").replace("_", " ")
+            )
+
+        examples.setdefault(outermost_parent, []).append(
+            {"path": example_path, "friendly_name": " / ".join(friendly_parts)}
+        )
+
+    return examples
+
+
+# Global context settings
+@app.context_processor
+def global_template_context():
+    with open("package.json") as package_json:
```

Comment:
is it worth opening every time we load a page the file? Maybe it could be done at the load of the application.

Reply:
Good idea. Fixing in https://github.com/canonical-web-and-design/vanilla-framework/pull/2807